src/backend/distributed/commands/table.c

```diff
@@ -138,6 +139,13 @@ PreprocessDropTableStmt(Node *node, const char *queryString,
 			continue;
 		}
 
+		if(PartitionTable(relationId))  {
+			Oid parentId = PartitionParentOid(relationId);
+			LockRelationOid(parentId, AccessExclusiveLock);
+			LockRelationOid(relationId, AccessExclusiveLock);
+			LockParentShardResourcesForShardsOfPartition(relationId);
```

Comment:
1) After the above 2 locks, we cannot have any other operation on the table, so LockParentShardResourcesForShardsOfPartition seems redundant.

2) Should these locks be taken on other metadata nodes as well?